components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/ThumbnailsUseCases.kt

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.thumbnails
+
+import android.graphics.Bitmap
+import mozilla.components.browser.state.selector.findTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.browser.thumbnails.storage.ThumbnailStorage
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * Contains use cases related to the thumbnails feature.
+ */
+class ThumbnailsUseCases(
+    store: BrowserStore,
+    thumbnailStorage: ThumbnailStorage
+) {
+    /**
+     * Load thumbnail use case.
+     */
+    class LoadThumbnailUseCase internal constructor(
+        private val store: BrowserStore,
+        private val thumbnailStorage: ThumbnailStorage
+    ) {
+        private val logger = Logger("ThumbnailsUseCases")
+
+        /**
+         * Loads the thumbnail of a tab from its in-memory [ContentState] or from the disk cache
+         * of [ThumbnailStorage].
+         */
+        suspend operator fun invoke(sessionIdOrUrl: String): Bitmap? {
+            val tab = store.state.findTab(sessionIdOrUrl)
```

Comment:
Thinking about this more, I wonder if we can find a way to request a screenshot as part of this method OR a new use case that would do that.

I can see it getting complicated though because we would need to pass a reference to the view in here..

Reply:
Follow up in https://github.com/mozilla-mobile/android-components/issues/7144